pkg/skaffold/yamltags/tags.go

```diff
@@ -74,6 +77,27 @@ func GetYamlTag(value interface{}) string {
 	return rawStr[:i]
 }
 
+// GetYamlTags returns the tags of the non-nested fields of the given non-nil value
+func GetYamlTags(value interface{}) []string {
+	var tags []string
+
+	buf, err := yaml.Marshal(value)
+	if err != nil {
+		logrus.Warnf("error marshaling %-v", value)
+		return tags
+	}
+
+	r := bufio.NewScanner(bytes.NewBuffer(buf))
+	for r.Scan() {
+		l := r.Text()
+		if !unicode.IsSpace(rune(l[0])) && l != "{}" {
```

Comment:
Can you explain what this does?
Probably would help if you add example on what the tag value looks like.
e.g. what is the json for input interface.
```
{yaml:"helm,omitempty" {yaml:"kubectl,omitempty"
```

Reply:
If ```value interface{}``` is 
```go
latest.DeployType{HelmDeploy: &HelmDeploy{...}, KustomizeDeploy: &KustomizeDeploy{...}}
```
then this parses that interface as it's raw yaml,
```yaml
kubectl:
  manifests:
  - k8s/*.yaml
kustomize:
  paths:
 - k8s/
```
and returns ["kubectl", "kustomize"], if latest.DeployType{} is passed or an interface in a similar manner than the raw yaml looks like ```{}```.

I tried using reflection to retrieve the tag name, but I could not get a method that generalized easily to values that were never nil.

Reply:
ah got it. Can you add this description in the code itself? just above the logic

Reply:
Done ✌